addons/l10n_in/models/account_invoice.py

```diff
@@ -148,3 +149,43 @@ def _l10n_in_get_hsn_summary_table(self):
                 'taxes_data': taxes_data,
             })
         return self.env['account.tax']._l10n_in_get_hsn_summary_table(base_lines, display_uom)
+
+    @api.depends('invoice_line_ids.tax_ids', 'l10n_in_gst_treatment', 'l10n_in_state_id.code')
+    def _compute_tax_type_warning(self):
+
+        def build_warning(record, action_name, message, views, domain=False):
+            return {
+                'message': message,
+                'action_text': _("View %s", action_name),
+                'action': record._get_records_action(name=_("Check %s") % action_name, target='current', views=views, domain=domain or [])
+            }
+
+        indian_invoice = self.filtered(lambda m: m.country_code == 'IN')
+        for move in indian_invoice:
+            if move.l10n_in_state_id.code == 'IN_OC' and move.l10n_in_gst_treatment == 'overseas' and move.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and line.tax_ids):
+                new_lines = lines = self.env['account.move.line']
+                for line in move.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and line.tax_ids):
+                    for tax in line.tax_ids:
+                        if tax.country_code == 'IN':
+                            l10n_in_tax_type = None
+                            if tax.env.ref('l10n_in.tax_tag_igst') in tax.invoice_repartition_line_ids.tag_ids:
+                                l10n_in_tax_type = 'igst'
+                        if l10n_in_tax_type != 'igst':
+                            lines |= line
+                for line in lines:
+                    if line._origin:
+                        new_lines |= line._origin
+
+                msg = _("Place of Supply is Foreign Country attracts IGST on Products")
+                move.l10n_in_tax_type_warning = {
+                    'invalid_gst_type_on_overseas_invoice': build_warning(
+                        message=msg,
+                        action_name=_("Journal Items(s)"),
+                        record=new_lines,
+                        views=[(self.env.ref("account.view_move_line_tree").id, "tree")],
+                        domain=[('id', 'in', new_lines.ids)]
+                    )
+                } if new_lines else {}
+            else:
+                move.l10n_in_tax_type_warning = {}
+        (self - indian_invoice).l10n_in_tax_type_warning = {}
```

Comment:
Firstly, I don't think we need to provide an action in this case. 
It is preferable that the user remain in the view where they can address the problem directly (rather than redirecting them elsewhere).

I've also simplified some of your conditions, but functionally they are the same. 

```suggestion
    @api.depends('invoice_line_ids.tax_ids', 'l10n_in_gst_treatment', 'l10n_in_state_id.code')
    def _compute_l10n_in_warnings(self):

        for move in self:
            if move.country_code != 'IN':
                move.l10n_in_warnings = {}
                continue

            warnings = {}
            if move.l10n_in_state_id.code == 'IN_OC' and move.l10n_in_gst_treatment == 'overseas':
                if any(
                    not self.env.ref('l10n_in.tax_tag_igst') in tax.invoice_repartition_line_ids.tag_ids
                    for tax in move.invoice_line_ids.tax_ids.filtered(lambda tax: tax.country_code == 'IN')
                ):
                    warnings.update({
                        'invalid_gst_type_on_overseas_invoice': {
                            'message': _("IGST should be used when the place of supply is a foreign country."),
                            'level': 'warning',
                        }
                    })

            move.l10n_in_warnings = warnings
```